src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -57,11 +67,32 @@ public Resource(IEnumerable<KeyValuePair<string, object>> attributes)
         /// </summary>
         public static Resource Empty { get; } = new Resource(Enumerable.Empty<KeyValuePair<string, object>>());
 
+        /// <summary>
+        /// Gets the dafault Resource with attributes from telemetry sdk  and OTEL_RESOURCE_ATTRIBUTES environment variable.
+        /// </summary>
+        public static Resource Default { get; } = new Resource(new List<KeyValuePair<string, object>>
+            {
+                new KeyValuePair<string, object>(TelemetrySdkNameKey, "opentelemetry"),
+                new KeyValuePair<string, object>(TelemetrySdkLanguageKey, "dotnet"),
+                new KeyValuePair<string, object>(TelemetrySdkVersionKey, Version.ToString()),
+            })
+            .Merge(GetOTelEnvVarResource());
+
         /// <summary>
         /// Gets the collection of key-value pairs describing the resource.
         /// </summary>
         public IEnumerable<KeyValuePair<string, object>> Attributes { get; }
 
+        /// <summary>
+        /// Creates a new <see cref="Resource"/> with user provided attributes, telemetry sdk attributes, and attributes from OTEL_RESOURCE_ATTRIBUTES environment variable.
+        /// </summary>
+        /// <param name="attributes">An <see cref="IDictionary{String, Object}"/> of attributes that describe the resource.</param>
+        /// <returns><see cref="Resource"/>.</returns>
+        public static Resource Create(IEnumerable<KeyValuePair<string, object>> attributes = null)
```

Comment:
Now we have a ctor() and a create() method both taking attributes as argument, each with different behavior. We should eliminate ctor() or this method. 
I don't think its correct to call a method Create(attributes) if its going to do more than just creating a resource with the provider attributes.

Reply:
How about calling it CreateWithDetectors() and pass a set of detectors to it. The method can then iterate through the detectors, each of them creates a resource and merge, and final output consist of a Resource consisting of all attributes from the detectors.
And by default, the TracerProviderSdk() ctor, can call userResource.Merge(CreateWithDetectors(default detectors)). default detectors can be "Library" and "EnvVar" detectors.

Reply:
btw, Go is doing this differently from java/python
Go is doing it differently. https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/resource/resource.go#L37

Reply:
From todays Sig meeting:
Eliminate Create(attr) or Ctor(attr) - this will just create Resource with the passed arguments.
For ENV/Library, introduce CreateFromDetectors(array[] detector){}.
For now, we *may* keep Detector as internal API, and later expose it as public API.
This SDK will ship with 2 builti-in detector - ENV,TelemetrySDK.